datadome-istio/files/datadome.lua

```diff
@@ -153,6 +153,36 @@ local function parse_xdd_header(value)
   return t
 end
 
+-- @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control
+local cacheable_directives = {
+	["max-age"] = true,
+	["x-maxage"] = true,
+	["public"] = true,
+}
+
+-- this returns whether the response can be cached
+-- according to its cache-control header value
+local function is_cacheable_response(response_handle)
+	local headers = response_handle:headers()
+
+	local cacheControlValue = headers:get("cache-control")
+
+	if not cacheControlValue then
+		return false
+	end
+
+	-- gmatch will split the string in order to extract each directive part
+	-- "public, max-age=604800, stale-if-error=86400" will become
+	-- public, max-age, 604800, stale-if-error, 86400
```

Comment:
Not the cleanest split. Ideally I would remove the `=xxxx` part from the loop but I haven't find how to do so with `gmatch` yet. Any idea?